src/agreement/aggr_agreement.rs

```diff
@@ -19,39 +20,30 @@ use super::{accumulator, verifiers};
 pub(super) async fn verify(
     ru: &RoundUpdate,
     committees_set: Arc<Mutex<CommitteeSet>>,
-    msg: &Message,
+    hdr: &Header,
+    aggr: &AggrAgreement,
```

Comment:
Would make sense to have this as &self and implementing this verify in AggrAgreement?